Framework/API/inc/MantidAPI/ISISInstrumentDataCache.h

```diff
@@ -0,0 +1,30 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2024 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+//----------------------------------------------------------------------
+// Includes
+//----------------------------------------------------------------------
+#include "MantidAPI/DllConfig.h"
+#include "MantidAPI/IArchiveSearch.h"
+#include <string>
+
+namespace Mantid {
+namespace API {
+
+class MANTID_API_DLL ISISInstrumentDataCache {
+public:
+  ISISInstrumentDataCache(const std::string &path) : m_dataCachePath(path) {}
+  std::string getFileParentDirectoryPath(std::string filename);
+
+private:
+  std::pair<std::string, std::string> validateInstrumentAndNumber(std::string filename);
+  std::pair<std::string, std::string> splitIntoInstrumentAndNumber(std::string filename);
```

Comment:
I think all these functions can also be `const`

Reply:
The reason I am not passing these as `const &` is because I am using the method `FileFinder::Instance()::extractAllowedSuffix()` on line https://github.com/mantidproject/mantid/pull/37187/files#diff-ab9dd41f9e44e9c2d5c363cc67efbd0e441a8322ac636636a86f68971f1cf935R47, which does take a reference and might change the string. I tried changing this method in the code before but it was causing other unexpected behaviours so I decided to instead pass the argument as a copy to keep this PR as localized as possible (to avoid changing older parts of the code). In the other function, `splitIntoInstrumentAndNumber`, I am using a `std::transform` inside the function, which alters the string argument. I could change this by passing the argument as `const &` and then creating an internal copy, and finally performing the transform. Do you think that would be preferred?

Reply:
I can add const to these methods, yes :+1:

Reply:
Ah I see, thanks for explaining. I think it would be preferred to create an internal copy in each of these methods

Reply:
I see, just did that :+1: